src/gui/core/timer.cpp

```diff
@@ -37,73 +40,35 @@ struct timer
 };
 
 /** Ids for the timers. */
-static size_t next_timer_id = 0;
+static std::atomic<size_t> next_timer_id(0);
```

Comment:
Doesn't need to be atomic. `timers_mutex` can (and should IMO) protect this variable.

Reply:
You might need `volatile` then?